### PR TITLE
volume/csi-wrapper: support multiple mount path in a single container

### PR DESCRIPTION
**Pull request by lysliu**
volume/csi-wrapper: support multiple mount path

Fixes: https://github.com/confidential-containers/cloud-api-adaptor/issues/1077

### CODE REVIEW

**Comment by huoqifeng** [comment]
@lysliu thanks for bring the fix, would you please correct the commit message following https://github.com/confidential-containers/cloud-api-adaptor/blob/main/.github/workflows/commit-message-check.yaml.

File: pkg/forwarder/interceptor/interceptor.go [hunk 1]
```diff
@@ -103,12 +104,14 @@ func (i *interceptor) CreateContainer(ctx context.Context, req *pb.CreateContain
 					logger.Printf("Failed to create dir: %v", err)
 				}
 			}
-
-			if isTargetPath(m.Source, volumeTargetPath) {
-				logger.Printf("Waiting for device mounted to: %s", m.Source)
-				err := waitForDeviceMounted(ctx, m.Source)
-				if err != nil {
-					return nil, err
+			volumeTargetPathSlice := strings.Split(volumeTargetPath, ",")
```
Comment (genjuro214) [inline pkg/forwarder/interceptor/interceptor.go:107]: It's better to split the slice outside the for, and calling strings.Trim() to clean the spaces before calling isTargetPath

File: pkg/adaptor/proxy/service.go [hunk 2]
```diff
@@ -104,11 +104,14 @@ func (s *proxyService) CreateContainer(ctx context.Context, req *pb.CreateContai
...
-		for _, m := range req.OCI.Mounts {
+		for i, m := range req.OCI.Mounts {
 			logger.Printf("        destination:%s source:%s type:%s", m.Destination, m.Source, m.Type)
 
 			if isNodePublishVolumeTargetPath(m.Source, kataDirectVolumesDir) {
-				req.OCI.Annotations[volumeTargetPathKey] = m.Source
+				if i > 0 {
+					req.OCI.Annotations[volumeTargetPathKey] += ", "
+				}
+				req.OCI.Annotations[volumeTargetPathKey] += m.Source
```
Comment (genjuro214) [inline pkg/adaptor/proxy/service.go:114]: why not add "," after adding m.Source? Any difference for the result?
```
req.OCI.Annotations[volumeTargetPathKey] += m.Source
req.OCI.Annotations[volumeTargetPathKey] += ","
```
and no need to add space, I think

Comment (lysliu) [reply pkg/adaptor/proxy/service.go:114]: add "," after adding m.Source will lead the size +1 of array `strings.Split(volumeTargetPath, ",")`, with a empty element

**Comment by genjuro214** [comment]
Need squash the commits and fix commit format like below (avoid long line):
```
volume/csi-wrapper: replay and update yaml files

fixes #1042

Signed-off-by: Lei Li <genjuro214@gmail.com>
```

**Comment by genjuro214** [comment]
@lysliu , pls fix the lint error

**Review by genjuro214 — APPROVED** [review]
LGTM

**Review by liudalibj — APPROVED** [review]
I didn't get time to test this pr, but the code changes looks good to me.
Thanks @lysliu !

**Comment by liudalibj** [comment]
@kartikjoshi21 @iaguis @bpradipt let you know that with this PR, we can support multiple mount path from pvcs in a single container

@lysliu @genjuro214 it's better that you add some document/sample for this feature in a new pr, thanks.

**Pull request merged**